src/sql/connection/connection.py

```diff
@@ -712,18 +712,18 @@ def dialect(self):
     def driver(self):
         return self._driver
 
-    def _connection_execute(self, query, parameters=None):
+    def _connection_execute(self, query, parameters={}):
```

Comment:
default function parameters should not be set this way, because Python will re-use the same dictionary across calls, which can lead to shared stated.

`None` is the way to go here. any reason why you did this change?

Reply:
Previously, when named_parameters was `false`, `parameters` = None. Then when named_parameters was `true`, `parameters` = `user_ns`, a non-empty dict.

I made this change so that we can add a third option in as `disable`, and this order seemed the most logical:

- When named_parameters is `disabled`, `parameters` = None.
- When named_parameters is `false`, `parameters` = {} (an empty dict). This is the default.
- When named_parameters is `true`, `parameters` = `user_ns` (a non-empty dict).

Because the `false` option was the default, I changed the default for parameters to be `{}`.

To avoid this issue I can change it to this:
- When named_parameters is `disabled`, `parameters` = {} (an empty dict).
- When named_parameters is `false`, `parameters` = None
- When named_parameters is `true`, `parameters` = `user_ns` (a non-empty dict).

Reply:
mutable objects should never be passed as default values in Python functions. [see this](https://medium.com/geekculture/do-not-use-objects-as-default-arguments-in-python-1c940212db2e)

whatever logic you're implementation, find an alternative approach because passing `parameters={}` will have undesirable side-effects